### PR TITLE
DBZ-627 Unifying TopicSelector implementations

### PR DESCRIPTION
**Pull request by gunnarmorling**
https://issues.jboss.org/browse/DBZ-627

I'll send the corresponding Oracle connector change tomorrow. I've decided to do this now in order to simplify a heartbeat implementation for PG and the others subsequently.

### CODE REVIEW

**Comment by jpechane** [comment]
@gunnarmorling Why was MongoDB connector ommitted?

**Comment by gunnarmorling** [comment]
Ah, good spot. I missed it because MongoDB's selector doesn't extend the
existing TopicSelector interface from core. I'll add that, too. Thanks!

**Comment by gunnarmorling** [comment]
@jpechane pushed one more commit to cover MongoDB, too.

File: debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/MongoDbTopicSelector.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    /**
+     * Get the name of the topic for a given prefix, database name, and collection name. This method returns
+     * "{@code <prefix><delimiter><dbName><delimiter><collectionName>}", and does not use the replica set name.
+     *
+     * @param delimiter the string delineating the prefix, database, and collection names; may not be null
+     * @param prefix the prefix; be null or empty if no prefix is needed
+     * @return the topic selector; never null
+     */
+    private static String getTopicName(CollectionId collectionId, String prefix, String delimiter) {
+        StringJoiner sb = new StringJoiner(delimiter);
```
Comment (jpechane) [inline debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/MongoDbTopicSelector.java]: @gunnarmorling  Is this really needed? Is prefix really optional?

**Comment by gunnarmorling** [comment]
It was like that before and I tried to leave the actual implementation
unchanged.

**Comment by jpechane** [comment]
@gunnarmorling Applied, thanks

**Pull request merged**